### PR TITLE
Fixing Front end issues related to image loading and SEO

### LINKED ISSUE #204: Fix Default Logo & also backoff if logo does not load
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

If a logo link breaks in the JSON (either the link is outdated, or there is a typo in the link), the website will continue to to request a logo from this URL:

https://community-scripts.github.io/logo.png

The script will continually pull until the user is rate blocked by github and cannot access the website. 

### Please provide a description of the expected behavior.

A default logo is shown, no logo will be requested more than once. If it doesn't load, show a placeholder. 

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

Screenshot 2024-11-13 at 1 33 31 PM

### Please provide detailed steps to reproduce the issue.

Visit a script page, such as:

https://community-scripts.github.io/ProxmoxVE/scripts?id=bookstack

Then using dev tools change the logo address to a url that will 404, then watch in the console as requests are made to https://community-scripts.github.io/logo.png

### LINKED ISSUE #205: Add canonical links
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

The website does not currently have rel="canonical" link tags in the head. It is a good idea to add this so Google and other search engines will not index forks.

### Please provide a description of the expected behavior.

Each page on the website has a link rel="canonical" link tag pointing to the official site (https://community-scripts.github.io/ProxmoxVE/)

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

Here is Google documentation on the issue:

https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls

### Please provide detailed steps to reproduce the issue.

Visit any page and inspect source.

### PR DESCRIPTION
**Pull request by bramsuurdje**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description

This PR fixes isses #205 and #204 

Fixes # (issue)

## Type of change
Please check the relevant option(s):

- [x] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [ ] Testing performed (I have tested my changes, ensuring everything works as expected)
- [ ] Documentation updated (I have updated any relevant documentation)

## Additional Information (optional)
Provide any additional context or screenshots about the feature or fix here.

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
- Related PR #205 and #204

### CODE REVIEW

**Review by MickLesk — DISMISSED** [review]

File: frontend/src/app/scripts/_components/ScriptAccordion.tsx [hunk 1]
```diff
@@ -104,13 +104,13 @@ export default function ScriptAccordion({
                   >
                     <div className="flex items-center">
                       <Image
-                        src={script.logo || "/logo.png"}
+                        src={script.logo || "/ProxmoxVE/logo.png"}
```
Comment (havardthom) [inline frontend/src/app/scripts/_components/ScriptAccordion.tsx:107]: Don't we have a `basePath` variable we can use here?

**Review by havardthom — DISMISSED** [review]
Should probably test the changes before merging

**Comment by MickLesk** [comment]
@BramSuurdje test & merge after Test 😀

File: frontend/src/app/scripts/_components/ScriptAccordion.tsx (hunk 1 above)
Comment (BramSuurdje) [reply frontend/src/app/scripts/_components/ScriptAccordion.tsx]: totally forgot, fixed.

**Review by havardthom — DISMISSED** [review]

**Comment by BramSuurdje** [comment]
when setting the URL of an image to a invalid one it now correctly fails over.

image

**Review by MickLesk — APPROVED** [review]

**Review by havardthom — APPROVED** [review]

**Comment by havardthom** [comment]
Tip: If you use the following syntax, issues will be closed automatically after merge

Fixes https://github.com/community-scripts/ProxmoxVE/issues/204
Fixes https://github.com/community-scripts/ProxmoxVE/issues/205

**Pull request merged**